### PR TITLE
[Fix] Portfolio main window buy/sell

### PR DESCRIPTION
**Pull request by BrettCleary**
# Summary

Refactor to share logic with overlay webview webcontents open window handler

need to merge `update_mm_ext` first and then https://github.com/HyperPlay-Gaming/extension-importer/pull/7

closes https://github.com/HyperPlay-Gaming/product-management/issues/783

# Testing

this test but for the main window portfolio page https://app.qase.io/case/HC-371

### CODE REVIEW

**Review by flavioislima — APPROVED** [review]
lgtm

File: src/backend/storeManagers/storeManagerCommon/games.ts [hunk 1]
```diff
@@ -70,49 +70,81 @@ const openBlankWindow = (options: BrowserWindowConstructorOptions) => {
   return browserGame.webContents
 }
 
+function handleUrlOpenForWebContents(
+  url: string,
+  contents: Electron.WebContents,
+  /* eslint-disable-next-line */
+  extensionImporter: any
```
Comment (jiyuu-jin) [inline src/backend/storeManagers/storeManagerCommon/games.ts:77]: Why the any type here?

Comment (BrettCleary) [reply src/backend/storeManagers/storeManagerCommon/games.ts:77]: It is an optional submodule :sweat_smile: `@hyperplay/extension-importer`

File: src/backend/__mocks__/electron.ts [hunk 2]
```diff
@@ -16,17 +16,20 @@ const dialog = {
...
     return '1.0.0'
-  },
-  requestSingleInstanceLock: jest.fn().mockReturnValue(true)
+  }
+
+  requestSingleInstanceLock = jest.fn().mockReturnValue(true)
 }
 
+const app = new App()
+
```
Comment (eliobricenov) [inline src/backend/__mocks__/electron.ts:32]: why did you need to do this?

Comment (BrettCleary) [reply src/backend/__mocks__/electron.ts:32]: `app.on('web-contents-created', openNewBrowserGameWindowListener)` needs to be mocked or the tests fail

app inherits from `EventEmitter` in electron

**Review by eliobricenov — APPROVED** [review]

**Review by nyghtstalker — APPROVED** [review]
Confirmed that the user is can now successfully perform a purchaser using the client (not just the overlay).
